### PR TITLE
feat: add features to example group chat

### PR DESCRIPTION
**Pull request by gbogboadePush**
- Update group info in realtime
- Add send image messages in example group chat
- Add promote to admin, demote to speaker

### CODE REVIEW

**Review by madhur-push — CHANGES_REQUESTED** [review]
Realtime messages aren't showing up on the UI, I can see them in the sockets log though.

File: example/lib/views/account_provider.dart [hunk 1]
```diff
@@ -162,8 +165,14 @@ class AccountProvider extends ChangeNotifier {
     // To get group creation or updation events
     pushSDKSocket.on(EVENTS.CHAT_GROUPS, (groupInfo) {
       print('CHAT NOTIFICATION EVENTS.CHAT_GROUPS: $groupInfo');
+      final type = (groupInfo as Map)['eventType'];
+
+      if (type == 'request') {
+        ref.read(requestsProvider).loadRequests();
```
Comment (madhur-push) [inline example/lib/views/account_provider.dart]: @Gbogboade1 We are getting the request here itself, we dont have to call loadRequests() imo, we just need to put that requests in the list of requests.

**Review by madhur-push — CHANGES_REQUESTED** [review]
The requests aren't reflecting in realtime.

File: example/lib/views/account_provider.dart (hunk 1 above)
Comment (Gbogboade1) [reply example/lib/views/account_provider.dart]: The api call is done to ensure the list is loaded with the most accurate data...other items in the list  may no longer exist

Comment (madhur-push) [reply example/lib/views/account_provider.dart]: Ayo, lets say you receive a request, the only way in which this request will come out of the requests list is when you will accept it correct? What im trying to say is the only way in which a request will get removed from the requests list is when a user will accept it, and the way in which a request will get added is via the sockets. We cant call the requests API after every socket event reason being what if there are many many requests right, in that case we'll unnecessarily exhaust the BE.

One thing we need to check is lets say I have a request and my request gets cancelled then in that case, is a socket event getting received or no. Lets make a ticket for the last thing I mentioned.

**Review by madhur-push — APPROVED** [review]
LGTM

**Pull request merged**